infrastructure/ansible/roles/ats_sim/tasks/ats_sim.yml

```diff
@@ -0,0 +1,66 @@
+---
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+- name: Install Node.js
+  yum:
+    name: "nodejs"
+    state: installed
+    update_cache : yes
+    lock_timeout: 300
+    enablerepo: "{{ additional_yum_repos | default([]) }}"
+
+- name: Create install directory
+  file:
+    path: "{{sim_install_dir}}"
+    state: directory
+    owner: root
+    group: root
+
+- name: Copy environment file
+  template:
+    src: "env.j2"
+    dest: "{{ sim_install_dir }}/env"
+    owner: root
+    group: root
+    mode: 0644
+
+- name: Copy unit file
+  template:
+    src: "simulator.service.j2"
+    dest: "/lib/systemd/system/simulator.service"
+    owner: root
+    group: root
+    mode: 0555
+
+- name: Git ats_sim.js file
+  git:
+    repo: "{{ sim_git_repo }}"
+    dest: /tmp/tc
+    version: "{{ sim_git_version }}"
+    accept_hostkey: yes
+
+- name: Copy Javascript file to intall directory
+  copy:
+    remote_src: yes
+    src: "/tmp/tc/test/astats_sim/ats_sim.js"
+    dest: "{{ sim_install_dir }}"
+    owner: root
+    group: root
+    mode: 0644
```

Comment:
Consider wrapping in a block using a when clause based on the existence of `sim_git_repo` and `sim_git_version`.

Reply:
Also consider adding a raw string mode as an alternate way to lay down this file for greater accessibility when git repos may not be available.  The ats simulator is a bit weird because it doesn't have an RPM like every other ATC component.

Reply:
> Also consider adding a raw string mode as an alternate way to lay down this file for greater accessibility when git repos may not be available. The ats simulator is a bit weird because it doesn't have an RPM like every other ATC component.

I'm don't know what 'raw string mode' is. If you are suggesting that I store the script in the lab manager or somewhere we can retrieve it if we're restoring an previous version of the deployment then maybe I should just keep it as a file in the ats_sim role itself. Akward but not as weird.

Reply:
Actually. It really doesn't make sense to pull this from Git at all if we allow for any kind of version identifier other than a commit hash since that is the only way to guarantee the same state during a 'restore' operation.

Reply:
Yeah, it's a bit of a weird bird since this is an ATC component that doesn't have an RPM.  Supporting git is a reasonable way to distribute it on average, but in the event that the source isn't OSS for some reason and because it's a single JS file; you could also just as easily support using the copy module with a content argument as an alternative.  This would permit the content from coming from anywhere.

Reply:
The standard position I've taken is that it's out of scope to define where component "binaries" come from in a generic core.  I view these as optional helpers because the component has no standard distribution vehicle.